### ISSUE TITLE
[Bug]: Type initializer for 'ReactiveUI.RxApp' throws TypeLoadException (DispatcherScheduler)

### ISSUE THREAD

**Issue by codedevote** [reactiveui/ReactiveUI#3448]
### Describe the bug 🐞

Hi,

I am maintaining a WPF project (.NET 4.7.2), which is currently on ReactiveUI and ReactiveUI.Wpf version 9.22.1 and using System.Reactive version 4.3.2. I didn't change anything compared to the last successful build/release (which is ~8 months ago). I now need to add some minor bugfixes, but I am no longer able to compile the solution. I get an exception:  

System.TypeInitializationException: 'The type initializer for 'ReactiveUI.RxApp' threw an exception.'  
Inner Exception:
TypeLoadException: Could not load type 'System.Reactive.Concurrency.DispatcherScheduler' from assembly 'System.Reactive, Version=4.3.0.0, Culture=neutral, PublicKeyToken=94bc3704cddfc263'.

I tried a lot of things, also copied an implementation of DispatcherScheduler to my project, but the exception remains. I have to say, that I am using paket for package management and I see that it always adds references to netcore versions of System.Reactive. But even if I manually replace those with net46 assembly, the exception remains. I don't have an idea how to proceed with that at the moment. I thought, maybe I drop paket and switch to nuget (which was improved over the years, so not sure, if all arguments to use paket will still hold) in first instance. But this is quite a big monolith having around 70 projects, so this is some work, which I would only do, if there is a chance that it works afterwards. Sorry for not being able to provide more details, please request any missing information, I am happy to provide that.

### Step to reproduce

Currently only happening in my solution, so no repro.

### Expected behavior

Not even sure, if ReactiveUI isn't behaving as expected and it's just my fault, it doesn't work.

### IDE

Visual Studio 2022, Visual Studio 2019

### Operating system

Windows

### Version

11

### Device

PC

### ReactiveUI Version

9.22.1

**Comment by dpvreony** [comment]
check your app.config to see if you have an assembly binding redirect for System.Reactive that is pointing to 4.3.0 as you have mentioned you're on 4.3.2 and that version of ReactiveUI requires 4.1.6. if that's the issue then the redirect should read something like

```
            <bindingRedirect oldVersion="4.1.6.0-4.3.2.0.0" newVersion="4.3.2"/> 
```

https://learn.microsoft.com/en-us/dotnet/framework/configure-apps/file-schema/runtime/bindingredirect-element

**Comment by codedevote** [comment]
Good catch, the app.config had a binding redirect to 4.3.0, but even after changing that, the error persists. DispatcherScheduler class is not available - I verified by trying to set RxApp.MainThreadScheduler = DispatcherScheduler.Current.  
Any other ideas? I still suspect paket to add wrong dependencies, but I cannot build even with setting the dependency manually. As I was able to build the same thing 8 months ago, has there been any change in VS that would cause such issue? Sorry, if I cannot provide better error details at the moment, I am still confused why it won't just work.

**Comment by glennawatson** [comment]
If it's the DispatcherScheduler feels almost like paket is not including the windows system.reactive package. 

I converted a company recently from paket to nuget with 1000 projects. The way I did it is read the paket.lock/paket.references file using the paket.core nuget package. Then using Microsoft.build nuget package I used msbuild to add PackageReference. You can then just delete the paket files.

**Comment by codedevote** [comment]
@glennawatson Thanks for the hint to these two nuget packages. I was able to hack sth together that was converting paket.references to PackageReferences., I also converted the .NET Framework 4.7.2 projects to .NET 7 and I am now also able to use the latest bits of ReactiveUI again. It was quite some work to convert all ReactiveList occurrences to DynamicData, as I found some things behaving differently, but all in all I am quite happy that everything is working again. Thanks a lot for your help!
